src/libnnstreamer-edge/nnstreamer-edge-internal.c

```diff
@@ -931,10 +876,12 @@ _nns_edge_send_thread (void *thread_data)
 
           conn_data = (nns_edge_conn_data_s *) eh->connections;
           while (conn_data) {
-            /** @todo update code for each connect type */
             conn = conn_data->sink_conn;
-            _nns_edge_transfer_data (conn, data_h, conn_data->id);
-
+            ret = _nns_edge_transfer_data (conn, data_h, conn_data->id);
+            if (NNS_EDGE_ERROR_NONE != ret) {
+              nns_edge_loge ("Failed to transfer data. Close the connection.");
+              _nns_edge_remove_connection_by_conn (eh, conn);
```

Comment:
Suggestion: 
1. _nns_edge_remove_connection_by_conn() is unnecessary.
2. Need fix assertion case (set next connection data before releasing it)

```
client_id = conn_data->id;
conn = conn_data->sink_conn;
ret = _nns_edge_transfer_data (conn, data_h, client_id);
conn_data = conn_data->next;

if (NNS_EDGE_ERROR_NONE != ret) {
  nns_edge_loge ("Failed to transfer data. Close the connection.");
   _nns_edge_remove_connection (eh, client_id);
}
```

Reply:
Thanks! Updated.